server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcBasePlugin.java

```diff
@@ -42,29 +44,73 @@
 
+    /*
+        At the moment, when writing into some table, the table name is concatenated with a special string
+        that is necessary to write into HDFS. However, a raw table name is necessary in case of JDBC.
+        This Pattern allows to extract the correct table name from the given RequestContext.dataSource
+     */
+    private static final Pattern tableNamePattern = Pattern.compile("/(.*)/[0-9]*-[0-9]*_[0-9]*");
+
+    // '100' is a recommended value: https://docs.oracle.com/cd/E11882_01/java.112/e16548/oraperf.htm#JJDBC28754
+    private static final int DEFAULT_BATCH_SIZE = 100;
+    private static final int DEFAULT_POOL_SIZE = 1;
+
+    // configuration parameter names
+    private static final String JDBC_DRIVER_PROPERTY_NAME = "jdbc.driver";
+    private static final String JDBC_URL_PROPERTY_NAME = "jdbc.url";
+    private static final String JDBC_USER_PROPERTY_NAME = "jdbc.user";
+    private static final String JDBC_PASSWORD_PROPERTY_NAME = "jdbc.password";
+
+    // DDL option names
+    private static final String JDBC_DRIVER_OPTION_NAME = "JDBC_DRIVER";
+    private static final String JDBC_URL_OPTION_NAME = "DB_URL";
+
+    // JDBC parameters from config file or specified in DDL
+    private String jdbcDriver, jdbcUrl, jdbcUser, jdbcPassword;
+
+    protected String tableName = null;
+
+    // After argument parsing, this value is guaranteed to be >= 1
+    protected int batchSize = DEFAULT_BATCH_SIZE;
+    protected boolean batchSizeIsSetByUser = false;
+
+    // Thread pool size
+    protected int poolSize = DEFAULT_POOL_SIZE;
+
+    // Quote columns setting set by user (three values are possible)
+    protected Boolean quoteColumns = null;
+
+    // Columns description
+    protected List<ColumnDescriptor> columns = null;
+
+    private static final Logger LOG = LoggerFactory.getLogger(JdbcBasePlugin.class);
+
     @Override
     public void initialize(RequestContext context) {
         super.initialize(context);
 
-        jdbcDriver = context.getOption("JDBC_DRIVER");
-        if (jdbcDriver == null) {
-            throw new IllegalArgumentException("JDBC_DRIVER is a required parameter");
-        }
+        // process configuration based params that could be auto-overwritten by user options
+        jdbcDriver = configuration.get(JDBC_DRIVER_PROPERTY_NAME);
+        assertMandatoryParameter(jdbcDriver, JDBC_DRIVER_PROPERTY_NAME, JDBC_DRIVER_OPTION_NAME);
+
+        jdbcUrl = configuration.get(JDBC_URL_PROPERTY_NAME);
+        assertMandatoryParameter(jdbcUrl, JDBC_URL_PROPERTY_NAME, JDBC_URL_OPTION_NAME);
 
-        dbUrl = context.getOption("DB_URL");
-        if (dbUrl == null) {
-            throw new IllegalArgumentException("DB_URL is a required parameter");
+        // authentication information is not required
+        jdbcUser = configuration.get(JDBC_USER_PROPERTY_NAME);
+        if (jdbcUser != null) {
+            jdbcPassword = configuration.get(JDBC_PASSWORD_PROPERTY_NAME);

```

Comment:
`getpassword` for jceks support?
```suggestion
            jdbcPassword = configuration.getPassword(JDBC_PASSWORD_PROPERTY_NAME);
```

Reply:
this would require converting returned byte[] to String and handling possible exception -- let's do it when we officially support the password encryption feature ?

Reply:
yeah, makes sense